Docs: run vela-core locally for debugging & breakpoints (#697)
oam-dev/kubevela.io: docs/contributor/code-contribute.md

make e2e-test
```

> ### Debug & breakpoints

Reviewer: ```suggestion
### Debugging Locally with Remote KubeVela Environment
```

> To run vela-core locally for debugging and breakpoints

Reviewer: ```suggestion
To run vela-core locally for debugging with kubevela installed in the remote cluster:
```

- Firstly, scaling the replicas of `kubevela-vela-core` to 0 for leader election of `controller-manager`, e.g `kubectl scale deploy -n vela-system kubevela-vela-core --replicas=0`.
- Secondly, removing the `WebhookConfiguration`, otherwise an error will be reported when applying your application using `vela-cli` or `kubectl`.
  ```shell
  Internal error occurred: failed calling webhook 'validating.core.oam.dev.v1beta1.applications': Post "https://vela-core-webhook.vela-system.svc:443/validating-core-oam-dev-v1beta1-applications?timeout=10s"
  ```

> - Finally, pulling the repository of KubeVela and debugging the code with breakpoints.

Reviewer: ```suggestion
- Finally, you can use the commands in the above [Build](#build) and [Testing](#Testing) sections, such as `make run`, to code and debug in your local machine.
```

* [kubevela.io Developer Guide](https://github.com/kubevela/kubevela.io/blob/main/README.md)
* [Update KubeVela.io CLI Reference Doc](./cli-ref-doc)

Great, you have complete the lifecycle of code contribution, try to [join the community as a member](https://github.com/kubevela/community/blob/main/community-membership.md) if you're interested.

## FAQ

### Q: How can I run vela-core locally for debugging?
 - Firstly, scaling the replicas of `kubevela-vela-core` to 0 for leader election of `controller-manager`, e.g `kubectl scale deploy -n vela-system kubevela-vela-core --replicas=0`.
 - Secondly, removing the `WebhookConfiguration`, otherwise an error will be reported when applying your application using `vela-cli` or `kubectl`.
   ```shell
   Internal error occurred: failed calling webhook 'validating.core.oam.dev.v1beta1.applications': Post "https://vela-core-webhook.vela-system.svc:443/validating-core-oam-dev-v1beta1-applications?timeout=10s"
   ```

>  - Finally, pulling the repository of KubeVela and debugging the code with breakpoints. 

Reviewer: What's the difference between the current make vela-core run in the section https://kubevela.io/docs/contributor/code-contribute#build

Author: > This part should be in the code contribution gude

ok

Author: > What's the difference between the current make vela-core run in the section https://kubevela.io/docs/contributor/code-contribute#build

build is about how to run locally, it is mainly about how to debug locally
